msprime/simulations.py

```diff
@@ -1338,14 +1338,30 @@ def __repr__(self):
         return repr(self.__dict__)
 
 
+def matrix_exponential(A):
+    """
+    Returns the matrix exponential of A.
+    https://en.wikipedia.org/wiki/Matrix_exponential
+    Note: this is not a general purpose method and is only intended for use within
+    msprime.
+    """
+
+    d, Y = np.linalg.eig(A)
+    Yinv = np.linalg.pinv(Y)
+    D = np.diag(np.exp(d))
+    B = np.matmul(Y, np.matmul(D, Yinv))
+    return B
+
+
 class DemographyDebugger(object):
     """
     A class to facilitate debugging of population parameters and migration
     rates in the past.
     """
     def __init__(
             self, Ne=1, population_configurations=None, migration_matrix=None,
-            demographic_events=[], model="hudson"):
+            demographic_events=[], model='hudson'):
```

Comment:
I think this is not how this default value should be specified: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

perhaps: `demographic_events=None` and then
```
if demographic_events is None:
    demographic_events = []
```